Jenkinsfile

```diff
@@ -65,7 +62,48 @@ node ("boost && fedora") {
             junit 'test/unit_tests_run.xml'
             failure_function(e, 'Tests failed')
         }
+
+        try {
+            stage("Build docs") {
+                sh "make html"
+                if (env.BRANCH_NAME != 'master') {
+                  archiveArtifacts artifacts: 'doc/build/'
+                }
+          }
+        } catch (e) {
+            failure_function(e, 'Docs generation failed')
+        }
     }
 
+    dir("docs") {
+        try {
+            stage("Publish docs") {
+                checkout scm
+
+              if (env.BRANCH_NAME == 'master') {
+                sh "git config user.email 'dm-jenkins-integration@esss.se'"
+                sh "git config user.name 'cow-bot'"
+                sh "git config remote.origin.fetch '+refs/heads/*:refs/remotes/origin/*'"
 
+                sh "git fetch"
+                sh "git checkout gh-pages"
+                sh "shopt -u dotglob && rm -rf ./*"
+                sh "mv -f ../build/doc/build/* ./"
+                sh "git add -A"
+                sh "git commit -a -m 'Auto-publishing docs from Jenkins'"
```

Comment:
Not strictly necessary, but the Jenkins variable `$BUILD_NUMBER` could be used in the commit message to identify the number of the build that made the commit.